consensus/misc/eip1559/eip1559.go

```diff
@@ -132,6 +132,53 @@ func ValidateHoloceneExtraData(extra []byte) error {
 	return ValidateHolocene1559Params(extra[1:])
 }
 
+// DecodeMinBaseFeeExtraData decodes the extraData parameters from the encoded form defined here:
+// https://github.com/ethereum-optimism/design-docs/blob/main/protocol/minimum-base-fee.md#minimum-base-fee-in-block-header
+//
+// Returns 0,0,0 if the format is invalid, though ValidateMinBaseFeeExtraData should be used instead of this function for
+// validity checking.
+func DecodeMinBaseFeeExtraData(extra []byte) (uint64, uint64, uint8) {
+	// Best effort to decode the extraData for every block in the chain's history,
+	// including blocks before the minimum base fee feature was enabled.
+	if len(extra) == 9 {
+		// This is Holocene extraData
+		denominator, elasticity := DecodeHolocene1559Params(extra[1:9])
+		return denominator, elasticity, 0
+	} else if len(extra) == 10 {
+		// This is the case when minBaseFeeLog2 feature is enabled
+		denominator := binary.BigEndian.Uint32(extra[1:5])
+		elasticity := binary.BigEndian.Uint32(extra[5:9])
+		minBaseFeeLog2 := extra[9]
+		return uint64(denominator), uint64(elasticity), minBaseFeeLog2
+	}
+	return 0, 0, 0
+}
+
+// EncodeMinBaseFeeExtraData encodes the EIP-1559 and minBaseFeeLog2 parameters into the header 'ExtraData' format.
+// Will panic if EIP-1559 parameters are outside uint32 range.
+func EncodeMinBaseFeeExtraData(denom, elasticity uint64, minBaseFeeLog2 uint8) []byte {
+	r := make([]byte, 10)
+	if denom > gomath.MaxUint32 || elasticity > gomath.MaxUint32 {
+		panic("eip-1559 parameters out of uint32 range")
+	}
+	r[0] = 1
+	binary.BigEndian.PutUint32(r[1:5], uint32(denom))
+	binary.BigEndian.PutUint32(r[5:9], uint32(elasticity))
+	r[9] = minBaseFeeLog2
+	return r
+}
+
+// ValidateMinBaseFeeExtraData checks if the header extraData is valid according to the minimum base fee feature.
+func ValidateMinBaseFeeExtraData(extra []byte) error {
```

Comment:
i think this can be refactored to be called `ValidateExtraData` to accommodate the best effort of validating the Holocene and Jovian cases. 

For example in here: https://github.com/ethereum-optimism/optimism/blob/develop/op-program/client/l2/engineapi/l2_engine_api.go#L364